### PR TITLE
Fix job cli create issue

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
Fixes JOB CLI create issue

1. First issue:
```
nvflare job create -force -j "./jobs/peft_p-tuning_local_345M" -w sag_pt_deploy_map -sd "code"\
   -f meta.conf min_clients=3 \
   -f app_1/config_fed_client.conf app_script="megatron_gpt_peft_tuning.py" app_config="max_steps=10000 val_check_interval=100" \
   -f app_2/config_fed_client.conf app_script="megatron_gpt_peft_tuning.py" app_config="max_steps=10000 val_check_interval=100" \
   -f app_3/config_fed_client.conf app_script="megatron_gpt_peft_tuning.py" app_config="max_steps=10000 val_check_interval=100" \
   -f app_server/config_fed_server.conf num_rounds=1
```

This does not work, will see

```
`Unable to handle command: create due to: Please specify one of the app names ['app_server', 'app_1', 'app_2']. For example '<app_name>/xxx.conf k1=v1 k2=v2...' `
```

2. Want to copy more file formats other than just files with postfix ".py" with the "-sd" options.

### Description

- in order to follow original logic, introduce a special "META_APP" just for meta file
- FILE_POSTFIX includes a list of supported postfixes

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/tool/job/config/configer.py [hunk 1]
```diff
@@ -288,10 +299,12 @@ def build_config_file_indices(job_folder: str, app_names: List[str]) -> Dict[str
     return app_config_file_index
 
 
-def get_app_name_from_path(path):
+def get_app_name_from_path(path: str):
     # path is in the format of as app1/xxx.conf
     # path app1/xxx.conf
     # xxx.conf
+    if _is_meta_file(path):
```
Comment (SYangster) [inline nvflare/tool/job/config/configer.py:306]: Here the path could include the directory, while `_is_meta_file()` currently assumes just a filename

Comment (YuanTingHsieh) [reply nvflare/tool/job/config/configer.py:306]: good catch, let me fix this

File: nvflare/tool/job/job_client_const.py [hunk 2]
```diff
@@ -43,8 +43,12 @@
...
 DEFAULT_APP_NAME = "app"
+# for consistency, meta config have a dummy app name
+META_APP_NAME = "__meta__app__"
 APP_CONFIG_DIR = "config"
 
 APP_SCRIPT_KEY = "app_script"
 APP_CONFIG_KEY = "app_config"
 TEMPLATES_KEY = "templates"
+
+FILE_POSTFIX = [".py", ".json", ".jsonl", ".yml", ".yaml", ".conf"]
```
Comment (holgerroth) [inline nvflare/tool/job/job_client_const.py:54]: Should we restrict this? What if sd contains checkpoint or binary files?

Comment (YuanTingHsieh) [reply nvflare/tool/job/job_client_const.py:54]: @holgerroth do you want to copy over checkpoint or binary files?
Even if we copy all the thing here in job creation, in job submission we still have some constraint on what files can be submitted right?
@IsaacYangSLA @yhwen Do you remember what constraint we have?

Comment (YuanTingHsieh) [reply nvflare/tool/job/job_client_const.py]: remove the restriction here in job creation, up to fl client/server side to accept / reject jobs

**Review by SYangster — APPROVED** [review]
LGTM, thanks

**Pull request merged**